cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/LoadPersistedConcepts.kt

```diff
@@ -0,0 +1,318 @@
+                mapper.readValue<PersistedConceptsEntry>(file)
+            } catch (ex: Exception) {
+                logger.error(
+                    "Error reading persisted concepts from ${file.path}: ${ex.message}",
+                    ex,
+                )
+                return
+            }
+
+        entries.concepts?.let { concepts ->
+            concepts.forEach { concept ->
+                if (concept.signature != null && concept.location != null) {
+                    log.error(
+                        "Both signature and location are set. Please use only one of them. The entire entry will be ignored!"
+                    )
+                    return@forEach
+                } else if (concept.signature != null) {
+                    getNodesBySignature(translationResult, concept.signature).forEach {
+                        underlyingNode ->
+                        addConcept(underlyingNode, concept.concept)
+                    }
+                } else if (concept.location != null) {
+                    getNodesByLocation(translationResult, concept.location).forEach { underlyingNode
+                        ->
+                        addConcept(underlyingNode, concept.concept)
+                    }
+                } else {
+                    log.error(
+                        "Neither signature nor location are set. The entire entry will be ignored!"
+                    )
+                }
+            }
+        }
+    }
+
+    /**
+     * This function retrieves the nodes matching the provided [signature] from the
+     * [translationResult]. Currently, this function only matches on the node being a
+     * [CallExpression] and the [CallExpression.reconstructedImportName] matching the provided FQN.
+     *
+     * @param translationResult The [TranslationResult] to search for nodes.
+     * @param signature The [SignatureEntry] containing the signature to match.
+     * @return A list of nodes matching the provided [SignatureEntry].
+     */
+    private fun getNodesBySignature(
+        translationResult: TranslationResult,
+        signature: SignatureEntry,
+    ): List<Node> {
+        return translationResult.getCallsByFQN(signature.fqn)
+    }
+
+    /**
+     * This function retrieves the nodes matching the provided [location] from the
+     * [translationResult].
+     *
+     * @param translationResult The [TranslationResult] to search for nodes.
+     * @param location The [LocationEntry] containing the location to match.
+     * @return A list of nodes matching the provided [LocationEntry] or an empty list of the
+     *   [location] cannot be parsed.
+     */
+    private fun getNodesByLocation(
+        translationResult: TranslationResult,
+        location: LocationEntry,
+    ): List<Node> {
+        val loc =
+            try {
+                val regex =
+                    Regex(
+                        "(?<startLine>\\d+):(?<startColumn>\\d+)-(?<endLine>\\d+):(?<endColumn>\\d+)"
```

Comment:
Is there an existing parser? Otherwise, we could consider specifying the attributes `startLine`, `startColumn`, `endLine`, and `endColumn` as part of the yml. This would remove parsing the string here and it would also help clarify the format for the user (the numbers and the cryptic format are probably more error-prone)

Reply:
I like the condensed format

Reply:
I can live with either of them